test_runner/fixtures/utils.py

```diff
@@ -499,3 +499,26 @@ def __repr__(self) -> str:
 
     def __str__(self) -> str:
         return f"'aux-{self.value}'"
+
+
+class PropagatingThread(threading.Thread):
+    _target: Any
+    _args: Any
+    _kwargs: Any
+    """
+    Simple Thread wrapper with join() propagating the possible exception in the thread.
+    """
+
+    def run(self):
+        self.exc = None
+        try:
+            self.ret = self._target(*self._args, **self._kwargs)
+        except BaseException as e:
+            log.info(f"oh fuck exception! {e}")
+            self.exc = e
+
+    def join(self, timeout=None):
```

Comment:
Does this re-raising not happen by default?

There is the ThreadPoolExecutor which does communicate the exception via the future/promise/whatever api.

Reply:
> Does this re-raising not happen by default?

No :(

> There is the ThreadPoolExecutor which does communicate the exception 

Yep, it would work, but the interface is getting even heavier: create thread pool, submit job, wait for it, terminate thread pool. So I'd leave as is.